### PR TITLE
Allow empty geometries

### PR DESCRIPTION
**Pull request by m-mohr**
As discussed yesterday, empty geometries can occur after certain opeations and as such we should try to handle them.

Implements https://github.com/Open-EO/openeo-processes/issues/404

Also aligns the dimension names with the API spec a bit more.

### CODE REVIEW

File: documentation/1.0/datacubes.md [hunk 1]
```diff
@@ -19,12 +19,12 @@ It is important to understand that datacubes are designed to make things easier
 A vector datacube on the other hand could look like this:
 
 <figure>
-    <img src="./datacubes/vector.png" alt="An examplary vector datacube with 3 dimensions: 2 geometries are given for the `Geometries` dimension, along with 3 timesteps for the temporal dimension `time` and 4 bands in the `Bands` dimension.">
-    <figcaption>An examplary vector datacube with 3 dimensions: 2 geometries are given for the <code>Geometries</code> dimension, along with 3 timesteps for the temporal dimension <code>time</code> and 4 bands in the <code>Bands</code> dimension.</figcaption>
+    <img src="./datacubes/vector.png" alt="An examplary vector datacube with 3 dimensions: 2 geometries are given for the `geometries` dimension, along with 3 timesteps for the temporal dimension `time` and 4 bands in the `bands` dimension.">
+    <figcaption>An examplary vector datacube with 3 dimensions: 2 geometries are given for the <code>geometries</code> dimension, along with 3 timesteps for the temporal dimension <code>time</code> and 4 bands in the <code>bands</code> dimension.</figcaption>
```
Comment (mkadunc) [inline documentation/1.0/datacubes.md:23]: Could we change the figure to align with "usually named `geometry`"? It's weird to have examples that don't follow the recommended naming.

Comment (m-mohr) [reply documentation/1.0/datacubes.md:23]: Yeah, would be nice. I don't have the capacity for it right now as it's Paint fiddling for me as this has been extracted from another image... I'll open a separate issue as this is also "wrong" in other images.

Comment (m-mohr) [reply documentation/1.0/datacubes.md:23]: https://github.com/Open-EO/openeo.org/issues/71

**Review by mkadunc — APPROVED** [review]
lgtm

File: documentation/1.0/datacubes.md [hunk 2]
```diff
@@ -56,15 +56,15 @@ For example, temporal labels are usually encoded as [ISO 8601](https://en.wikipe
...
+A geometry dimension is not included in the example raster datacube above and it is not used in the following examples, but to show how a vector dimension with two polygons could look like:
 
 | name       | type   | labels | reference system |
 | ---------- | ------ | ------ | ---------------- | 
 | `geometry` | vector | `POLYGON((-122.4 37.6,-122.35 37.6,-122.35 37.64,-122.4 37.64,-122.4 37.6))`, `POLYGON((-122.51 37.5,-122.48 37.5,-122.48 37.52,-122.51 37.52,-122.51 37.5))` | [EPSG:4326](https://epsg.io/4326) |
 
 A dimension with geometries can consist of points, linestrings, polygons, multi points, multi linestrings, or multi polygons.
 It is not possible to mix geometry types, but the single geometry type with their corresponding multi type can be combined in a dimension (e.g. points and multi points).
-Empty geometries (includes GeoJSON `null` geometries) are not allowed.
+Empty geometries (includes GeoJSON `null` geometries) are allowed and can sometimes also be the result of certain vector operations such as a negative buffer.
```
Comment (soxofaan) [inline documentation/1.0/datacubes.md:67]: > includes GeoJSON `null` geometries

this looks very cryptic to me, I'm not sure what to make from it. 
Is it about a GeoJSON Feature where the "geometry" member is `null`?

**Review by soxofaan — COMMENTED** [review]
minor remark

File: documentation/1.0/datacubes.md (hunk 2 above)
Comment (m-mohr) [reply documentation/1.0/datacubes.md:67]: You mean the part in brackets?

Maybe:
```suggestion
Empty geometries (including GeoJSON Features with a `null` geometry or GeoJSON geometries with empty coordinates) are allowed and can sometimes also be the result of certain vector operations such as a negative buffer.
```

Comment (soxofaan) [reply documentation/1.0/datacubes.md:67]: indeed

minor tweak:
```suggestion
Empty geometries (such as GeoJSON Features with a `null` geometry or GeoJSON geometries with an empty coordinates array) are allowed and can sometimes also be the result of certain vector operations such as a negative buffer.
```

Comment (m-mohr) [reply documentation/1.0/datacubes.md]: Done

**Review by soxofaan — APPROVED** [review]

**Pull request merged**